src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -467,15 +470,32 @@ private void RenderUnboundAttribute(string attributeName, Chunk attributeValueCh
                     // Dynamic attribute value should be run through the conditional attribute removal system. It's
                     // unbound and contains C#.
 
+                    // TagHelper attribute rendering is buffered by default. We do not want to write to the current
+                    // writer.
+                    var currentTargetWriter = _context.TargetWriterName;
+                    var currentWriteAttributeMethodName = _context.Host.GeneratedClassContext.WriteAttributeValueMethodName;
+                    _context.TargetWriterName = null;
+
+                    Debug.Assert(attributeValueChunk is ParentChunk);
+                    var children = ((ParentChunk)attributeValueChunk).Children;
+                    var attributeCount = children.Count(c => c is DynamicCodeAttributeChunk || c is LiteralCodeAttributeChunk);
+
                     _writer
-                        .WriteStartMethodInvocation(_tagHelperContext.AddHtmlAttributeValuesMethodName)
+                        .WriteStartMethodInvocation(_tagHelperContext.BeginAddHtmlAttributeValuesMethodName)
+                        .Write(ExecutionContextVariableName)
```

Comment:
Yeah. In general most of ours APIs tend to have least variable -> most variable going left to right. In this case the context changes the least (for a single attribute). Plus it also makes the start and end look a bit symmetrical

```C#
BeginAddHtmlAttributeValues(context, "attrname", 1);
EndAddHtmlAttributeValues(context);
```